src/map/unit.cpp

```diff
@@ -171,8 +171,13 @@ bool unit_walktoxy_nextcell(block_list& bl, bool sendMove, t_tick tick) {
 		return true;
 
 	// Reached end of walkpath
-	if (ud->walkpath.path_pos >= ud->walkpath.path_len)
+	if (ud->walkpath.path_pos >= ud->walkpath.path_len) {
+		// We need to send the reply to the client even if already at the target cell
+		// This allows the client to synchronize the position correctly
+		if (sendMove && bl.type == BL_PC)
```

Comment:
Maybe change the order for short-circuit. But I'm not sure if sendMove will be true more often than the other check.

Reply:
SendMove is a binary check so it should be faster. It's true when it's a new move command, but not true for any following cell moved. A PC moving happens also quite often, so... I don't think it makes a difference. My logic was that the binary check is faster than an integer comparison, but actually not so sure about it?

Reply:
Yeah I'm not sure it would make a huge difference, just maybe if this is checked more often for non-players. Anyway it was just a suggestion and probably not even relevant on modern compilers.